resources/bundles/org.eclipse.core.filesystem/src/org/eclipse/core/internal/filesystem/zip/ZipFileStore.java

```diff
@@ -0,0 +1,559 @@
+public class ZipFileStore extends FileStore {
+	/**
+	 * The path of this store within the zip file.
+	 */
+	private final IPath path;
+
+	/**
+	 * The file store that represents the actual zip file.
+	 */
+	private final IFileStore rootStore;
+
+	/**
+	 * Creates a new zip file store.
+	 */
+	public ZipFileStore(IFileStore rootStore, IPath path) {
+		this.rootStore = rootStore;
+		this.path = path.makeRelative();
+	}
+
+	private ZipEntry[] childEntries(IProgressMonitor monitor) throws CoreException {
+		List<ZipEntry> entryList = new ArrayList<>();
+		String myName = path.toString();
+
+		try (FileSystem zipFs = openZipFileSystem()) {
+			Path zipRoot = zipFs.getPath(myName);
+			Files.walkFileTree(zipRoot, EnumSet.noneOf(FileVisitOption.class), 1, new SimpleFileVisitor<Path>() {
+				@Override
+				public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) {
+					String entryName = zipRoot.relativize(file).toString();
+					if (!Files.isDirectory(file)) {
+						// For files, read attributes and create ZipEntry
+						ZipEntry zipEntry = new ZipEntry(entryName);
+						zipEntry.setSize(attrs.size());
+						zipEntry.setTime(attrs.lastModifiedTime().toMillis());
+						// Compressed size is not directly available; method is set based on ZIP standard
+						zipEntry.setMethod(ZipEntry.DEFLATED);
+						entryList.add(zipEntry);
+					} else {
+						// For directories, simply add them with a trailing slash
+						entryList.add(new ZipEntry(entryName + "/")); //$NON-NLS-1$
+					}
+					return FileVisitResult.CONTINUE;
+				}
+
+				@Override
+				public FileVisitResult preVisitDirectory(Path dir, BasicFileAttributes attrs) {
+					// Include directories only if they are not the root directory
+					if (!dir.equals(zipRoot)) {
+						String dirName = zipRoot.relativize(dir).toString() + "/"; //$NON-NLS-1$
+						entryList.add(new ZipEntry(dirName));
+					}
+					return FileVisitResult.CONTINUE;
+				}
+			});
+		} catch (IOException | URISyntaxException e) {
+			throw new CoreException(new Status(IStatus.ERROR, getPluginId(), "Error reading ZIP file", e)); //$NON-NLS-1$
+		}
+
+		return entryList.toArray(new ZipEntry[0]);
+	}
+
+	@Override
+	public IFileInfo[] childInfos(int options, IProgressMonitor monitor) throws CoreException {
+		ZipEntry[] entries = childEntries(monitor);
+		int entryCount = entries.length;
+		IFileInfo[] infos = new IFileInfo[entryCount];
+		for (int i = 0; i < entryCount; i++) {
+			infos[i] = convertZipEntryToFileInfo(entries[i]);
+		}
+		return infos;
+	}
+
+	@Override
+	public String[] childNames(int options, IProgressMonitor monitor) throws CoreException {
+		ZipEntry[] entries = childEntries(monitor);
+		int entryCount = entries.length;
+		String[] names = new String[entryCount];
+		for (int i = 0; i < entryCount; i++) {
+			names[i] = computeName(entries[i]);
+		}
+		return names;
+	}
+
+	/**
+	 * Computes the simple file name for a given zip entry.
+	 */
+	private String computeName(ZipEntry entry) {
```

Comment:
Can this be made static?

Reply:
made static in e528b7d4ea6f9572631fcd3b24d239f53323a342